app/src/main/java/com/talkable/demo/MainActivity.java

```diff
@@ -229,6 +236,34 @@ public void onDeepLinkingClick(View view) {
         if (offerId.length() > 0) {
             paramsMap.put(VISITOR_OFFER_KEY, offerId);
         }
+
+        updateTalkableCredentials();
+
         TalkableDeepLinking.track(paramsMap);
     }
+
+    private void updateTalkableCredentials() {
+        setServerIfNeeded();
+        setApiKeyAndSiteSlugIfNeeded();
+    }
+
+    private void setApiKeyAndSiteSlugIfNeeded() {
+        EditText siteSlugText = findViewById(R.id.siteSlugText);
+        String siteSlug = siteSlugText.getText().toString();
+        EditText apiKeyText = findViewById(R.id.apiKeyText);
+        String apiKey = apiKeyText.getText().toString();
+
+        if (siteSlug.length() > 0 && apiKey.length() > 0) {
+            Talkable.updateCredentials(apiKey, siteSlug);
+        }
+    }
+
+    private void setServerIfNeeded() {
+        EditText serverText = findViewById(R.id.serverText);
+        String server = serverText.getText().toString();
+
+        if (server == null && server.length() > 0) {
```

Comment:
Logic in this `if` looks like a bug. Also lets have just one method `updateTalkableCredentials` with all logic